src/datastore/postgres-store.ts

```diff
@@ -5538,17 +5542,21 @@ export class PgDataStore
         // join against `txs` to get the full transaction objects only for that page.
         `
         WITH stx_txs AS (
-          SELECT tx_id, ${countOverColumn()}
+          SELECT tx_id
           FROM principal_stx_txs
-          WHERE principal = $1 AND ${blockCond} AND canonical = TRUE AND microblock_canonical = TRUE
+          WHERE principal = $1 AND ${blockCond}
           ORDER BY block_height DESC, microblock_sequence DESC, tx_index DESC
           LIMIT $2
           OFFSET $3
         )
-        SELECT ${txColumns()}, ${abiColumn()}, count
+        SELECT ${txColumns()}, ${abiColumn()}, ${countOverColumn()}
         FROM stx_txs
-        INNER JOIN txs USING (tx_id)
-        WHERE canonical = TRUE AND microblock_canonical = TRUE
+        INNER JOIN txs
+          ON stx_txs.tx_id = txs.tx_id
+          AND txs.canonical = TRUE
+          AND txs.microblock_canonical = TRUE
+        LIMIT $2
+        OFFSET $3
```

Comment:
This is OK for a hotfix but I'm afraid performance for this query will be degraded significantly. I'd defer to you if you think this is something we need to deploy ASAP though.

Reply:
The LIMIT/OFFSET is still specified in the primary subquery (against the `principal_stx_txs` table), so I think this outer LIMIT/OFFSET will only be constraining the existing (small) set of results from the subquery. In this case, filtering results from the subquery which are not actually canonical due to a bug. 

I ran some local testing with one of the most active addresses, and actually saw a performance increase. That doesn't translate completely to production, but I don't think there will be a noticeable perf degradation.

Reply:
Leaving the LIMIT/OFFSET in the subquery as well as the outer query will lead to incorrect `total`s, because the outer `COUNT(*) OVER()::integer` will be counting rows from an already paginated subquery (i.e. only one page) instead of counting from the total number of transactions available to that principal.

This is why you're getting the test failures on the JSON `total` and also explains the performance increase you're seeing (i.e. the total txs you're joining against is already paginated so the query is very fast).

Reply:
Yeah, this PR will fix the "missing txs for an address" bug, but it will _not_ fix the `count`/pagination bug related to the same issue. Fixing that requires a change to the re-org ingestion which is a breaking change and will require a longer deployment cycle. I think for now having an incorrect count/total isn't as bad as the address endpoint missing txs